### ISSUE TITLE
Can't parse identifiers with Unicode escape sequence

### ISSUE THREAD

**Issue by ariya** [eslint/espree#181]
See ES2015 spec section 11.6.1, Unicode escape sequence can be used in identifier name.

Test case:

```js
espree.parse('var \\u{41}', { ecmaFeatures: { unicodeCodePointEscapes: true } })
```

Actual: Error: Line 1: Unexpected token ILLEGAL.

Expected: a valid syntax tree.

**Comment by nzakas** [comment]
Working on this.
